.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -306,9 +306,16 @@ private List<Token> DurationWithBeforeAndAfter(string text, DateObject reference
             foreach (var er in durationEr)
             {
                 // if it is a multiple duration and its type is not equal to Date than skip it.
-                if (er.Data != null && er.Data.ToString() != Constants.MultipleDuration_Date)
+                if (er.Data != null && er.Data.ToString().StartsWith("multipleDuration") && er.Data.ToString() != Constants.MultipleDuration_Date)
                 {
                     continue;
+                }
+
+                // special treatment for cases like "more than 3 days from today"
+                // should only extract "3 days from today" as datepoint
+                if (er.Data != null && (er.Data.ToString() == Constants.MORE_THAN_MOD || er.Data.ToString() == Constants.LESS_THAN_MOD))

```

Comment:
er = StripDurationInequality(er)
The if moves inside the renamed method.

Reply:
Fixed, here er is the iterative one, so create a "void" method doing similar work.